source/Halibut.TestUtils.CompatBinary.Base/Services/StayAliveUntilHelper.cs

```diff
@@ -0,0 +1,40 @@
+using System;
+using System.IO;
+using System.Runtime.InteropServices;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Halibut.TestUtils.SampleProgram.Base.Services
+{
+    public class StayAliveUntilHelper
+    {
+        public static void WaitUntilSignaledToDie()
+        {
+            var stayAliveFile = SettingsHelper.CompatBinaryStayAliveLockFile();
+            while (true)
+            {
+                try
+                {
+                    Console.WriteLine("Waiting to lock");
+                    var fileStreamLock = new FileStream(stayAliveFile, FileMode.Open, FileAccess.Read, FileShare.None);
+                    Console.WriteLine("Got a lock, going to die now");
+                    fileStreamLock.Dispose();
+                    File.Delete(stayAliveFile);
+                    Environment.Exit(0);
```

Comment:
Does `Environment.Exit` kill the test runner as well?

Reply:
This is an external process so no